src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlAnyElementAttributes.cs

```diff
@@ -17,61 +14,94 @@ namespace System.Xml.Serialization
     /// </devdoc>
     public class XmlAnyElementAttributes : IList
     {
-        private List<object> _list;
+        private List<XmlAnyElementAttribute> _list;
+
+        public XmlAnyElementAttributes()
+        {
+            _list = new List<XmlAnyElementAttribute>();
+        }
 
         /// <include file='doc\XmlAnyElementAttributes.uex' path='docs/doc[@for="XmlAnyElementAttributes.this"]/*' />
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
         public XmlAnyElementAttribute this[int index]
         {
-            get { return (XmlAnyElementAttribute)List[index]; }
-            set { List[index] = value; }
+            get { return _list[index]; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _list[index] = value;
+            }
         }
 
         /// <include file='doc\XmlAnyElementAttributes.uex' path='docs/doc[@for="XmlAnyElementAttributes.Add"]/*' />
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public int Add(XmlAnyElementAttribute attribute)
+        public int Add(XmlAnyElementAttribute value)
         {
-            return List.Add(attribute);
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            int index = _list.Count;
+            _list.Add(value);
+            return index;
         }
 
         /// <include file='doc\XmlAnyElementAttributes.uex' path='docs/doc[@for="XmlAnyElementAttributes.Insert"]/*' />
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public void Insert(int index, XmlAnyElementAttribute attribute)
+        public void Insert(int index, XmlAnyElementAttribute value)
         {
-            List.Insert(index, attribute);
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            _list.Insert(index, value);
         }
 
         /// <include file='doc\XmlAnyElementAttributes.uex' path='docs/doc[@for="XmlAnyElementAttributes.IndexOf"]/*' />
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public int IndexOf(XmlAnyElementAttribute attribute)
+        public int IndexOf(XmlAnyElementAttribute value)
         {
-            return List.IndexOf(attribute);
```

Comment:
The reason we have the List property is to implement some members of IList interface. So I prefer to work directly with the _list field for essential operations.